frontend/src/api/prometheus/__tests__/usePrometheusQueryRange.spec.ts

```diff
@@ -0,0 +1,121 @@
+import axios from 'axios';
+import { act } from '@testing-library/react';
+import { testHook } from '~/__tests__/unit/testUtils/hooks';
+import { mockPrometheusServing } from '~/__mocks__/mockPrometheusServing';
+import usePrometheusQueryRange from '~/api/prometheus/usePrometheusQueryRange';
+
+jest.mock('axios', () => ({
+  post: jest.fn(),
+}));
+
+const mockAxios = jest.mocked(axios.post);
+
+describe('usePrometheusQueryRange', () => {
+  const responsePredicate = jest.fn();
+  const queryLang = 'testQuery';
+  const span = 60;
+  const endInMs = 123456;
+  const step = 30;
+  const namespace = 'testNamespace';
+  const active = true;
+
+  it('should fetch data successfully and handle refresh ', async () => {
+    const mockedResponse = { data: { response: mockPrometheusServing({}) } };
+
+    mockAxios.mockResolvedValue(mockedResponse);
+    responsePredicate.mockReturnValue(mockedResponse);
+
+    const renderResult = testHook(usePrometheusQueryRange)(
+      active,
+      '/api/prometheus/serving',
+      queryLang,
+      span,
+      endInMs,
+      step,
+      responsePredicate,
+      namespace,
+    );
+
+    expect(renderResult).hookToStrictEqual([[], false, undefined, expect.any(Function), true]);
+
+    expect(mockAxios).toHaveBeenCalledTimes(1);
+    expect(renderResult).hookToHaveUpdateCount(1);
+
+    await renderResult.waitForNextUpdate();
+
+    expect(renderResult).hookToStrictEqual([
+      mockedResponse,
+      true,
+      undefined,
+      expect.any(Function),
+      false,
+    ]);
+    expect(mockAxios).toHaveBeenCalledTimes(1);
+    expect(renderResult).hookToHaveUpdateCount(2);
+
+    expect(renderResult).hookToBeStable([false, true, false, false, true]);
```

Comment:
This should have been apparent when writing the test when comparing the asserted previous and current values on line 39 and lines 47-51. `loading` value went from `false => true` and therefore there's no way the stability array index `1` could be `true` and pass the test. 

Fixed the bug in the comparative utility used by `hookToBeStable`: https://github.com/opendatahub-io/odh-dashboard/pull/2528

```suggestion
    expect(renderResult).hookToBeStable([false, false, false, true, false]);
```